torch/csrc/acc/Module.cpp

```diff
@@ -0,0 +1,205 @@
+  }
+
+  bool isAvailable() const override {
+    PYBIND11_OVERRIDE_PURE_NAME(
+        bool, at::PrivateUse1HooksInterface, "is_available", isBuilt, );
+  }
+
+  // TODO(qihqi): these is not supported from python yet
+  const at::Generator& getDefaultGenerator(
+      c10::DeviceIndex device_index) const override {
+    return at::PrivateUse1HooksInterface::getDefaultGenerator(device_index);
+  }
+
+  at::Generator getNewGenerator(
+      c10::DeviceIndex device_index = -1) const override {
+    return at::PrivateUse1HooksInterface::getNewGenerator(device_index);
+  }
+
+  at::Device getDeviceFromPtr(void* data) const override {
+    return at::PrivateUse1HooksInterface::getDeviceFromPtr(data);
+  }
+
+  bool isPinnedPtr(const void* data) const override {
+    return at::PrivateUse1HooksInterface::isPinnedPtr(data);
+  }
+
+  at::Allocator* getPinnedMemoryAllocator() const override {
+    return at::PrivateUse1HooksInterface::getPinnedMemoryAllocator();
+  }
+};
+
+struct PythonDeviceGuard final : public c10::impl::DeviceGuardImplInterface {
+  using c10::impl::DeviceGuardImplInterface::DeviceGuardImplInterface;
+
+  c10::DeviceType type() const override {
+    PYBIND11_OVERRIDE_PURE_NAME(
+        c10::DeviceType, c10::impl::DeviceGuardImplInterface, "type_", type, );
+  }
+
+  // TODO(qihqi): figure out if those are even useful
+  // to python or not
+  c10::Device exchangeDevice(c10::Device device) const override {
+    return getDevice();
+  }
+  c10::Device getDevice() const override {
+    return c10::Device(type(), 0);
+  }
+  void setDevice(c10::Device device) const override {}
+  void uncheckedSetDevice(c10::Device device) const noexcept override {}
+  c10::Stream getStream(c10::Device) const noexcept override {
+    // no-op
+    return c10::Stream(c10::Stream::DEFAULT, getDevice());
+  }
+
+  c10::Stream getNewStream(c10::Device, int priority = 0) const override {
+    // no-op
+    (void)priority;
+    return c10::Stream(c10::Stream::DEFAULT, getDevice());
+  }
+
+  c10::Stream exchangeStream(c10::Stream) const noexcept override {
+    // no-op
+    return c10::Stream(c10::Stream::DEFAULT, getDevice());
+  }
+  c10::DeviceIndex deviceCount() const noexcept override {
+    return 1;
+  }
+
+  // TODO(qihqi): support Event-related functions
+  void record(
+      void** /*event*/,
+      const c10::Stream& /*stream*/,
+      const c10::DeviceIndex /*device_index*/,
+      const c10::EventFlag /*flag*/) const override {}
+  void block(void* /*event*/, const c10::Stream& /*stream*/) const override {}
+  bool queryEvent(void* /*event*/) const override {
+    return true;
+  }
+  void destroyEvent(void* /*event*/, const c10::DeviceIndex /*device_index*/)
+      const noexcept override {}
+
+  // Stream-related functions
+  bool queryStream(const c10::Stream& /*stream*/) const override {
+    return true;
+  }
+  void synchronizeStream(const c10::Stream& /*stream*/) const override {}
+};
```

Comment:
Can we make these raise an error instead of being silently wrong?

Reply:
Not really. I need it to never fails, and that is the reason of overriding them here (it throws in the parent class). So either this don't fail OR, in Python I can override it to be no-op. Otherwise the python test wont pass.

> being silently wrong?

I'd love to understand why having these functions being no-op, especially when there is only one device, is considered "wrong".